pkg/certificate/providers/vault/certificate_manager_test.go

```diff
@@ -115,15 +173,15 @@ func TestIssueCertificate(t *testing.T) {
 
 }
 
-func mockVault(t *testing.T) (*vault.Core, string, string) {
+func mockVault(t *testing.T) (*vault.Core, string, string, net.Listener) {
 	coreConfig := &vault.CoreConfig{
 		LogicalBackends: map[string]logical.Factory{
 			"pki": pki.Factory,
 		},
 	}
 	core, _, rootToken := vault.TestCoreUnsealedWithConfig(t, coreConfig)
-	_, addr := vhttp.TestServer(t, core)
-	return core, rootToken, addr
+	ln, addr := vhttp.TestServer(t, core)
+	return core, rootToken, addr, ln
```

Comment:
since you're passing in t, you can do:


```
mockVault(...) (string, string) {
   ...
   t.Cleanup(func(){
       ln.Close()
       core.Shutdown()
   })
   return rootToken, addr
}
```